gsy_e_sdk/setups/Grid_Operator_API_scripts/rest_tou_strategy.py

```diff
@@ -0,0 +1,158 @@
+from gsy_e_sdk.utils import get_area_uuid_from_area_name_and_collaboration_id
+
+current_dir = os.path.dirname(__file__)
+
+MARKET_NAMES = [
+    "Grid",
+    "Community",
+]
+ORACLE_NAME = "dso"
+SLOT_LENGTH = 15  # leave as is
+AUTOMATIC = True
+
+
+class Oracle(Aggregator):
+    """Class to represent the Grid Operator client type."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.is_finished = False
+
+    def on_market_cycle(self, market_info):
+        current_market_fee = {}
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            if area_dict["area_name"] in MARKET_NAMES:
+                self.add_to_batch_commands.last_market_dso_stats(area_uuid)
+                current_market_fee[area_dict["area_name"]] = area_dict[
+                    "current_market_fee"
+                ]
+        self.execute_batch_commands()
+        next_market_fee = self.set_new_market_fee(market_info)
+        log_grid_fees_information(MARKET_NAMES, current_market_fee, next_market_fee)
+
+    def set_new_market_fee(self, market_info):
+        """Return the market fees for each market for the next time slot."""
+        next_market_fee = {}
+        market_time = from_format(market_info["market_slot"], DATE_TIME_FORMAT)
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            if area_dict["area_name"] in MARKET_NAMES:
+                next_market_fee[
+                    area_dict["area_name"]
+                ] = calculate_next_slot_market_fee(market_time, area_dict["area_name"])
+
+                self.add_to_batch_commands.grid_fees(
+                    area_uuid=area_uuid,
+                    fee_cents_kwh=next_market_fee[area_dict["area_name"]],
+                )
+        self.execute_batch_commands()
+        return next_market_fee
+
+    def on_event_or_response(self, message):
+        pass
+
+    def on_finish(self, finish_info):
+        self.is_finished = True
+
+
+def read_fee_strategy():
+    "Return a dictionary containing the Time of Use strategy loaded from the CSV input file."
+    with open(
+        os.path.join(current_dir, "resources/ToU.csv"), newline="", encoding="utf-8"
+    ) as csvfile:
+        csv_rows = csv.reader(csvfile, delimiter=" ", quotechar="|")
+        headers = next(csv_rows)[0].split(";")
+        market_indexes = {}
+        planned_fee = {}
+        for market_name in MARKET_NAMES:
+            market_indexes.update({(market_name, headers.index(market_name))})
+        for row in csv_rows:
+            row = row[0].split(";")
+            for market in MARKET_NAMES:
+                planned_fee.update({(row[0], market): row[market_indexes[market]]})
+    return planned_fee
+
+
+def calculate_next_slot_market_fee(market_time, market_name):
+    """Return the market fee for the next time slot."""
+    slot_time = (
+        market_time.add(minutes=1 * SLOT_LENGTH).time().format(TIME_FORMAT_SECONDS)
+    )
```

Comment:
Why do you need the additional `time()`? I don not think you need it as you are only exporting the hours and mintes and seconds with `format` already

Reply:
```suggestion
    slot_time = market_time.add(minutes=SLOT_LENGTH).format(TIME_FORMAT_SECONDS)
```